src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -110,7 +110,20 @@ EnsureReferenceTablesExistOnAllNodes(void)
 		return;
 	}
 
-	/* prevent this function from running concurrently with itself */
+	/*
+	 * Prevent this function from running concurrently with itself.
+	 *
+	 * It also prevents concurrent DROP TABLE or DROP SCHEMA. We need this
+	 * because through-out this function we assume values in referenceTableIdList
+	 * are still valid.
+	 *
+	 * We don't need to handle other kinds of reference table DML/DDL here, since
```

Comment:
We have isolation tests for these in `isolation_copy_placement_vs_modification`. We cannot test this with `replicate_reference_tables()` since `master_copy_shard_placement()` runs in a separate backend and isolation tester ignores locks in other backends.